### ISSUE TITLE
Report + Negotiation Card: Vet Report Content

### ISSUE THREAD

**Issue by turingcompl33t** [mlte-team/mlte#227]
Determine the content and organization that we want to provide in the report for the MLTE 0.2 release. We have spoken in the past regarding how the current content is merely a ripoff of model cards. If the content for the 0.2 release is just a reflection of the content in the negotiation card, so be it. Once the content and organization are known, inform @turingcompl33t to perform the updates.

**Comment by krmaffey** [comment]
# Report Structure for MLTE 0.2

## TLDR
I recommend we take creative license and follow option 3 below, using a combination of our own fields and model card fields organized in a way that makes sense for a non-ML user or a decision maker. All imports are optional, and once imported, all fields must be editable.

### Proposed MLTE 0.2 Report Structure

- Model Summary (Imports from ML problem type and ML task)
- Performance
    - Goals of the system (Imports from goals of the system)
    - MLTE evaluation (Imports from MLTE)
- Intended Use (Imports from usage context and production requirements)
- Risks (Imports from risk of FP, risk of FN, and other risks)
- Data (Imports from data source)
- Caveats and Recommendations (No import; free text field)
- Quantitative Analysis (Imports result graphs/tables from MLTE)

More details can be read below in my full evaluation.

## Structure Evaluation

For this issue, I am evaluating a few courses of action:
1. Use model cards
2. Use the negotiation card
3. Use model card fields mapped to negotiation card content
4. Use a completely unique configuration for the content

I am immediately eliminating approach 4 because we have no data to support that this makes sense for our current situation.

## 1. Model Cards

To evaluate this approach, I started by looking at a dummy MLTE report. This was not very useful so I found a model card for [GPT-3](https://github.com/openai/gpt-3/blob/master/model-card.md) and used that for the evaluation. Here are my thoughts:
- Having 'model details' at the outset of the model card seems misleading because rather than actual details, the section is used to give a high-level summary of the model.
- Listing the model date and version towards the top of the card seems inconsistent with an emphasis on giving the most relevant information at the top of the report.
- The rest of the card is informative but written in a lengthy and expert-focused way. Hard to judge if this is a result of the format or of the authors.

Next, I looked back at the actual model cards [paper](https://arxiv.org/pdf/1810.03993.pdf). The model card approach is very thorough, but seems to present information in a way that is logical for the developers rather than for a decision maker. Of note, I think that all the information they cover should be included in reporting on a model, just in a different format.

Conclusion: model cards are great content-wise, but I feel that they should be rearranged to be most useful for users and decision makers.

## 2. MLTE Negotiation Card

To review the negotiation card content, I used this [page](https://github.com/mlte-team/mlte/blob/master/mlte/negotiation/model.py) as well as a [list](https://docs.google.com/document/d/1xTicrKS4QrdeVV1mzgCTYrK4oAJ56UMy_8y-pcdn4T8/edit) in our google docs. My thoughts:
- It's important and useful to have the system goal and the metric to evalute its performance on that goal at the top of the card. This seems like it should be done similarly in a report.
- Most of the other fields correspond well to those used in model cards, with differences in terminology and organization.
- The terminology and structure we use in the negotiation card seems to lend itself to a discussion between parties rather than as a reporting mechanism.

Conclusion: some of the content from the negotiation card seems important for reporting, and some of it seems like it is too detailed to be important for an overall understanding of a ML system and how it performed on its main tasks.

## 3. Why Not Both?

This section looks at using the content of a negotiation card within the fields determined by [Model Cards](https://arxiv.org/pdf/1810.03993.pdf). Rather than a proper evaluation, this section is just me taking creative license based on what *seems* like a well-prioritized list of what someone would want to read in a model evaluation report.

- Model Summary
    - Instead of 'model details' from model cards, call this what it is: a summary. This is a field I made up, but is based on the way I saw some projects use model cards.
    - Offer users the opportunity to import text from the following sections:
        - ML problem type
        - ML task
    - Ensure the text is editable, especially if they do an import.
- Performance
    - This corresponds to model cards field 'metrics', renamed.
    - Goals of the system
        - Allow users to import from the negotiation card
    - MLTE evaluation
        - This should include the chosen metric to evaluate the how the model performs with respect to the system goal(s) as well as all the other measurements and evidence the team has chosen to collect throughout the evaluation process. Includes the baseline or threshold set by the development team for each metric.
        - If graphs are useful, they can be included here or in the Quantitative Analysis section at the bottom
- Intended Use
    - This is a model cards field
    - Allow users to import from the negotiation card for the following fields:
        - Usage context
        - Production requirements
- Risks
    - This is an adaptation of the model cards field 'factors'
    - Allow users to import from the three risk sections (risk of FP, risk of FN, other risks)
- Data
    - Model cards has two data sections - training data and evaluation data. Breaking these up is a potential future TODO once we get some user feedback.
    - Allow users to import from data source
- Caveats and Recommendations
    - Model cards field
    - Free text field
- Quantitative Analysis
    - Model cards field
    - Annex-like section allowing more space for graphs and tables.
    
Based on the above evaluation, I recommend we use this combined structure. I am replicating it in a simpler form at the top of this page for easier review.

**Comment by krmaffey** [comment]
@turingcompl33t and @JennyNiemann, if you're short on time feel free to just review the top of my report which contains a proposed report structure. Would appreciate getting comments from both of you before we proceed with this as the report structure for MLTE 0.2. Thank you!
